src/probnum/quad/bq_methods/stop_criteria/_stopping_criterion.py

```diff
@@ -0,0 +1,101 @@
+"""Stopping criteria for Bayesian quadrature."""
+
+from typing import Callable
+
+import numpy as np
+
+from probnum.quad.bq_methods.bq_state import BQState
+from probnum.randvars import Normal
+from probnum.type import FloatArgType, IntArgType
+
+
+class StoppingCriterion:
+    """Base class for a stopping criterion.
+
+    Parameters
+    ----------
+    stopping_criterion :
+        A function that determines whether a convergence criterion is reached.
+    """
+
+    def __init__(self, stopping_criterion: Callable):
+        self._stopping_criterion = stopping_criterion
```

Comment:
`stopping_criterion` argument not needed I suppose. Rather raise NotImplementedError in `__call__`

Reply:
Yes this is a weird construction that I copied from [here](https://github.com/JonathanWenger/probnum/blob/linsolve-refactor/src/probnum/linalg/solvers/stop_criteria/_stopping_criterion.py).
Why would the `__call__`  method be an argument of the class, @JonathanWenger ?

Reply:
This was originally intended to be a convenient way for a user to create a (custom) policy simply from a callable. This turned out to be bad design as you rightfully point out. We will simply force subclassing instead for all components of PN methods.

Reply:
done